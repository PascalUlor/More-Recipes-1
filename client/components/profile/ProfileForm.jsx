import React from 'react';
import PropTypes from 'prop-types';
import Spinner from 'react-md-spinner';
import TextFieldGroup from '../../common/TextFieldGroup.jsx';

const ProfileForm = (props) => {
  const {
    handleChange, handleFocus, handleClick, handleImageChange,
    handleSubmit, isUpdating
  } = props,
  {
    fullName, username, email, location,
    aboutMe, profileImage, errors
  } = props.profile;
  return (
    <div className="row">
      <div className="col-4 col-sm-4 col-md-4 col-lg-4 mt-1">
        <figure className="figure">
          <img src={profileImage} id="profile-image"
            onClick={handleClick} className="rounded-circle img-fluid" alt="Site Logo"/>
          <figcaption className="figure-caption text-center">
            <h6 className="mt-2">Profile Picture</h6>
          </figcaption>
        </figure>
        <input type="file" name="imageFile" className="file-chooser"
          onChange={handleImageChange} accept="image/*"/>
      </div>
      <div className="col-8 col-sm-8 col-md-8 col-lg-8">
        <h2>Basic Information</h2>
        <form role="form" onSubmit={handleSubmit}>
          <TextFieldGroup
            label='Full Name' font="fa fa-user"
            name='fullName' value={fullName}
            error={errors.fullName} onChange={handleChange}
            onFocus={handleFocus} placeholder='firstname lastname'/>
          <TextFieldGroup
            label='Username' font="fa fa-user"
            name='username' value={username}
            error={errors.username} onChange={handleChange}
            onFocus={handleFocus} placeholder='enter username'/>
          <TextFieldGroup
            label='Email' font="fa fa-envelope"
            name='email' value={email}
            error={errors.email} onChange={handleChange}
            onFocus={handleFocus} placeholder='enter email'/>
          <TextFieldGroup
            label='Location' font="fa fa-map-marker"
            name='location' value={location}
            error={errors.location} onChange={handleChange}
            onFocus={handleFocus} placeholder='enter city where you are. eg: lagos'/>
          <div className="form-group">
            <label htmlFor="about">About Me</label>
            <textarea rows="5" className="form-control"
              name= "aboutMe" value={aboutMe} onChange={handleChange}
              onFocus={handleFocus} placeholder="write about yourself">
            </textarea>
            {errors.aboutMe && <span className="text-danger small">{errors.aboutMe}</span>}
          </div>
          <button type="submit" className="btn btn-outline-info" disabled={isUpdating}>
           { !isUpdating ? 'Save Changes' : <span>Saving... <Spinner size={20} /></span> }
          </button>
        </form>
      </div>
    </div>
  );
};

ProfileForm.propTypes = {
    handleChange: PropTypes.func.isRequired,
    handleFocus: PropTypes.func.isRequired,
    handleClick: PropTypes.func.isRequired,
    handleImageChange: PropTypes.func.isRequired,
    profile: PropTypes.shape(),
    isUpdating: PropTypes.bool.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default ProfileForm;