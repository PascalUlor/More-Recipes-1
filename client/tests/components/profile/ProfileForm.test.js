/* eslint-disable */
import React from 'react';
import toJson from 'enzyme-to-json';
import Spinner from 'react-md-spinner';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileForm from '../../../components/profile/ProfileForm.jsx';

/**
 * @description return initial property state of the component instance
 * @function setup
 *
 * @return { object } props - properties of component instance
 */
const setup = () => {
  const props = {
    profile: {
      fullName: '',
      username: '',
      email: '',
      profileImage: '',
      location: '',
      aboutMe: '',
      errors: {},
    },
    isUpdating: false,
    handleChange: jest.fn(),
    handleFocus: jest.fn(),
    handleClick: jest.fn(),
    handleImageChange: jest.fn(),
    handleSubmit: jest.fn()
  };
  return props;
};

const props = setup();
describe('ProfileForm component', () => {

  it('renders form with five input fields and a button', () => {
    const shallowWrapper = shallow(<ProfileForm {...props}/>);
    expect(shallowWrapper.find('div').length).toBe(5);
    expect(shallowWrapper.find('form').length).toBe(1);
    expect(shallowWrapper.find('TextFieldGroup').length).toBe(4);
    expect(shallowWrapper.find('button').length).toBe(1);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders Spinner component, when isLoading is true', () => {
    const mountWrapper =
    mount(<Router><ProfileForm {...props} isUpdating={true}/></Router>);
    expect(mountWrapper.find(Spinner).length).toEqual(1);
  });
});
