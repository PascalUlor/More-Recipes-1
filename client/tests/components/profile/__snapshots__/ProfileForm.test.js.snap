// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileForm component renders form with five input fields and a button 1`] = `
<form
  onSubmit={[MockFunction]}
  role="form"
>
  <div
    className="row"
  >
    <div
      className="col-6 col-sm-3 col-md-4 col-lg-4 figure"
    >
      <figure
        className="figure"
      >
        <img
          alt="Site Logo"
          className="rounded-circle img-fluid"
          id="profile-image"
          onClick={[MockFunction]}
          src=""
        />
      </figure>
      <input
        accept="image/*"
        className="file-chooser"
        name="imageFile"
        onChange={[MockFunction]}
        type="file"
      />
    </div>
    <div
      className="col-10 col-sm-9 col-md-8 col-lg-6 profile-text"
    >
      <TextFieldGroup
        font="fa fa-user"
        label="Full Name"
        name="fullName"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="firstname lastname"
        type="text"
        value=""
      />
      <TextFieldGroup
        font="fa fa-user"
        label="Username"
        name="username"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="enter username"
        type="text"
        value=""
      />
      <TextFieldGroup
        font="fa fa-envelope"
        label="Email"
        name="email"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="enter email"
        type="text"
        value=""
      />
      <TextFieldGroup
        font="fa fa-map-marker"
        label="Location"
        name="location"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="enter city where you are. eg: lagos"
        type="text"
        value=""
      />
      <div
        className="form-group"
      >
        <label
          className="page-text"
          htmlFor="about"
        >
          About Me
        </label>
        <textarea
          className="form-control"
          name="aboutMe"
          onChange={[MockFunction]}
          onFocus={[MockFunction]}
          placeholder="write about yourself"
          rows="5"
          value=""
        />
      </div>
      <div
        className="text-right"
      >
        <button
          className="btn btn-outline-info"
          disabled={false}
          type="submit"
        >
          Save Changes
        </button>
      </div>
    </div>
  </div>
</form>
`;
